snap7/types.py

```diff
@@ -93,55 +83,79 @@ class WordLen(Enum):
     Counter = 0x1C
     Timer = 0x1D
 
+    @property
+    def ctype(self) -> Type[c_int8 | c_int16 | c_int32]:
+        map_: Dict[WordLen, Type[c_int8 | c_int16 | c_int32]] = {
+            WordLen.Bit: c_int16,
+            WordLen.Byte: c_int8,
+            WordLen.Word: c_int16,
+            WordLen.DWord: c_int32,
+            WordLen.Real: c_int32,
+            WordLen.Counter: c_int16,
+            WordLen.Timer: c_int16,
+        }
+        return map_[self]
+
+
+class Area(Enum):
+    PE = 0x81
+    PA = 0x82
+    MK = 0x83
+    DB = 0x84
+    CT = 0x1C
+    TM = 0x1D
 
-# Leave it for now
-S7WLBit = 0x01
-S7WLByte = 0x02
-S7WLChar = 0x03
-S7WLWord = 0x04
-S7WLInt = 0x05
-S7WLDWord = 0x06
-S7WLDInt = 0x07
-S7WLReal = 0x08
-S7WLCounter = 0x1C
-S7WLTimer = 0x1D
-
-# Server Area ID  (use with Register/unregister - Lock/unlock Area)
-# NOTE: these are not the same for the client!!
-srvAreaPE = 0
-srvAreaPA = 1
-srvAreaMK = 2
-srvAreaCT = 3
-srvAreaTM = 4
-srvAreaDB = 5
-
-server_areas = {
-    "PE": srvAreaPE,
-    "PA": srvAreaPA,
-    "MK": srvAreaMK,
-    "CT": srvAreaCT,
-    "TM": srvAreaTM,
-    "DB": srvAreaDB,
-}
+    def wordlen(self) -> WordLen:
+        if self == Area.TM:
+            return WordLen.Timer
+        elif self == Area.CT:
+            return WordLen.Counter
+        return WordLen.Byte
+
+
+class SrvArea(Enum):
+    """
+    NOTE: these values are DIFFERENT from the normal area IDs.
+    """
+
+    PE = 0
+    PA = 1
+    MK = 2
+    CT = 3
+    TM = 4
+    DB = 5
+
+
+class Block(Enum):
```

Comment:
If you make the Block class a subclass of [IntEnum](https://docs.python.org/3/library/enum.html#enum.IntEnum), the `ctype`  property could look like this:
```python3
@property
    def ctype(self) -> c_int:
        return c_int(self)
```